cpp/include/cudf/io/detail/json.hpp

```diff
@@ -51,4 +51,16 @@ void write_json(data_sink* sink,
                 json_writer_options const& options,
                 rmm::cuda_stream_view stream,
                 rmm::mr::device_memory_resource* mr);
+
+/**
+ * @brief Normalize single quotes to double quotes using FST
+ *
+ * @param inbuf Input device span buffer
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource to use for device memory allocation
+ */
+rmm::device_uvector<char> normalize_single_quotes(cudf::device_span<std::byte const> inbuf,
```

Comment:
Have you considered device_span<char const> as input?
I feel like it makes more sense to match the in/out types. Also, the only call site would get a lot simpler.

Reply:
After offline discussions, the parameter for the input raw JSON buffer is changed to a rvalue reference of type `device_uvector` so that we can have the option of directly doing something with the input buffer.